pkg/resourcegroup/controllers/resourcegroup/resourcegroup_controller.go

```diff
@@ -201,20 +188,29 @@ func (r *reconciler) endReconcilingStatus(
 	}()
 	select {
 	case <-finish:
-		newStatus.ObservedGeneration = generation
-		// Update conditions, if they've changed
-		newStatus.Conditions = updateCondition(newStatus.Conditions,
-			newReconcilingCondition(v1alpha1.FalseConditionStatus, FinishReconciling, finishReconcilingMsg))
 		newStatus.Conditions = updateCondition(newStatus.Conditions,
 			aggregateResourceStatuses(newStatus.ResourceStatuses))
 	case <-time.After(reconcileTimeout):
-		// Update conditions, if they've changed
-		newStatus.Conditions = updateCondition(newStatus.Conditions,
-			newReconcilingCondition(v1alpha1.FalseConditionStatus, ExceedTimeout, exceedTimeoutMsg))
 		newStatus.Conditions = updateCondition(newStatus.Conditions,
 			newStalledCondition(v1alpha1.TrueConditionStatus, ExceedTimeout, exceedTimeoutMsg))
 	}
 
+	// Updating the ObservedGeneration is probably a no-op, because the applier
+	// should have updated it already when it updated the spec & status.
+	// But update it here just for completeness.
+	// It's important to note that because there's only one ObservedGeneration
+	// field, it cannot be used to confirm that this controller has completed.
+	// it can only be used to confirm that the applier has completed its update.
+	newStatus.ObservedGeneration = generation
+
+	// Previously, the Reconciling condition toggled between True & False while
+	// this controller was reconciling, but this caused unnecessary churn and
+	// confusion, because it didn't reflect the reconciliation status of the
+	// managed objects in status.resourceStatuses. So now it's always set to
+	// False, for reverse compatibility, in case any clients were waiting it.
+	newStatus.Conditions = updateCondition(newStatus.Conditions,
+		newReconcilingCondition(v1alpha1.FalseConditionStatus, FinishReconciling, finishReconcilingMsg))
```

Comment:
Should we set the reason to "Obsolete" or "Deprecated" instead of "FinishReconciling" to avoid confusion?

Reply:
@janetkuo suggests to keep the "FinishReconciling" reason.